substrate/dex/pallet/src/lib.rs

```diff
@@ -154,11 +154,34 @@ pub mod pallet {
   #[pallet::storage]
   pub type Pools<T: Config> = StorageMap<_, Blake2_128Concat, PoolId, PoolInfo<Coin>, OptionQuery>;
 
+  #[pallet::storage]
+  #[pallet::getter(fn last_quote_for_block)]
+  pub type LastQuoteForBlock<T: Config> =
+    StorageDoubleMap<_, Identity, BlockNumberFor<T>, Identity, Coin, [u8; 8], ValueQuery>;
+
+  /// Moving window of oracle prices.
+  #[pallet::storage]
+  #[pallet::getter(fn oracle_prices)]
+  pub type OraclePrices<T: Config> = StorageMap<_, Identity, [u8; 8], u16, OptionQuery>;
+  impl<T: Config> Pallet<T> {
+    /// Get the highest sustained(lowest) value of the window
+    pub fn highest_sustained_price() -> Amount {
+      let mut iter = OraclePrices::<T>::iter_keys();
+      // we can unwrap because we will always have at least 1 key in this map
+      // before this function is called.
```

Comment:
I really don't like this reasoning. It sets bounds on callers which aren't documented. Mind having this return an Option and having the caller handle it

OR

saying it's fine because the genesis enters a price of 0 for block 0?

Either make the assumptions endogenous, not exogenous.